### ISSUE TITLE
[iPad] Shapes sounds don't play until other buttons are pressed 

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/quadrilateral#429]
**Test device**
iPad 9th generation 

**Operating System**
16.4.1

**Browser**
safari

**Problem description**
For https://github.com/phetsims/qa/issues/926, with iPad + safari:
When I open Quad for the first time, the Shape Sounds don't play when I move a corner or edge. I have to press something else (ex. "Show Shape Name") to initiate the sounds. This only happens the first time I open the sim. I do not see this in dev.76. I also don't see this on mac + safari or iPad + chrome.

**Steps to reproduce**
1. In iPad settings press "Clear history and Website Data" under Safari
2. Open Quad
3. Move sides and corners

**Visuals**

https://user-images.githubusercontent.com/87318828/232504468-455ca02c-6b6d-4e82-b9a0-0edbf4b2f777.mov

<details>
Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Quadrilateral‬
URL: https://phet-dev.colorado.edu/html/quadrilateral/1.0.0-rc.2/phet/quadrilateral_all_phet.html
Version: 1.0.0-rc.2 2023-04-04 16:04:39 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36
Language: en-US
Window: 1600x780
Pixel Ratio: 1.7999999523162842/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 31 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {}
</details>

**Comment by jessegreenberg** [comment]
Wow, @Nancy-Salpepi I am very impressed you found it was related to clearing history and settings. Thank you for finding this.

Unfortunately, I cannot reproduce this in https://phet-dev.colorado.edu/html/quadrilateral/1.0.0-rc.2/phet/quadrilateral_all_phet.html on my iPad Air 2 iOS 15.0.2, even after clearing history and website data.

@jbphet I am wondering if you have seen something like this before?

**Comment by Nancy-Salpepi** [comment]
I also so this with JT, GFLB, Friction, and RAP(with query parameter screens=1) on master using the original steps I reported. 
I was not able to reproduce this with published sims. 


Also, if other tabs were open, and sound had been working in those tabs before going to master and opening one of the mentioned sims, returning to that tab resulted in no sound. 
Steps:
1. clear history & website data for safari in settings. 
2. you will be asked if you want to close tabs--select yes
3. Go to the phet website and open Friction--if will have sound
4. In a new tab, go to phettest and open Quad or JT --no sound
5. Go back to the first tab--Friction will now have no sound

Perhaps this is an iOS16 issue? @jessegreenberg you mentioned you were on 15.

**Comment by jessegreenberg** [comment]
OK thanks @Nancy-Salpepi, that is really helpful. Yes, it sounds like a new platform specific issue, but since you could not reproduce on master perhaps there is a tambo change that is causing this.

@jbphet can you please review this report, does this seem like something we can resolve?

**Comment by jbphet** [comment]
There is a pretty long history of us (mostly me) working on issues related to iPad sound not starting, or stopping at inopportune times.  It's hard to say without some investigation what is happening to cause the symptoms described above.  I'll bring this issue up as a possibility for the next sprint and we can go from there.

**Comment by jbphet** [comment]
I did some investigation on this because I thought it might be important for the upcoming release of the quadrilateral sim.  I was able to duplicate the problem on an iPad that is available to me, and I tried it first on iOS 14.4, and was not initially able to reproduce the problem, then upgraded the device to 16.4.1 and the *could* reproduce it.  However, it doesn't happen every time on 16.4.1, so I don't feel confident that this is entirely new in the 16.4.1 release, since I didn't really spend a lot of time trying to make it happen on 14.4.

I added some debug output to the screen so I could check the state of the audio context, since I suspected this was at the root of the problem, and this turned out to be true.  The audio context is starting in the `suspended` state, which it generally does on a device where it doesn't know whether a website can be trusted.  This is part of the "auto-play policy".  However, from the debug output, I can see the attempts that our sound manager makes to "resume" the audio context in response to a `touchstart` event isn't working.  From looking at the output in conjunction with the code, I can tell that the Promises returned by the attempts to resume the context aren't resolving at all.  That's why in the image below we see four lines at the bottom that indicate that the state has transitioned to `running` - because it apparently queued up all of the callbacks until the state change was allowed.  In this particular test case, I did several drags of the sides of the shape which didn't produce sound, and then just tapped the screen in a random spot, and sound started playing.

![image](https://user-images.githubusercontent.com/3082831/236582815-b7299a8f-1959-4c71-84c5-5ff0e8ad231f.png)

My educated guess on this is that Apple changed something subtle in the auto-play behavior and it no longer allows sound to be played based on just a `touchstart` event.  Maybe people were finding that they brushed the device accidently and sound started, so Apple thought they should require a more obviously deliberate interaction.  Hard to say.  I did some online searching on the topic and didn't find anything.

The bottom line is that there is no easy fix for this.  A full investigation would involve contacting Apple and potentially logging a bug with them, which would take some time, since I'd need to create simple example cases to demonstrate the problem.  I'm assigning this back to @jessegreenberg to work with the designer(s) and decide whether to escalate this or just live with it.  Since it seems to impact multiple sims, an escalation is probably needed.

**Comment by jbphet** [comment]
In case we want to revive the debug code that I used to print out the state info about, here's a patch.

<details>

Subject: [PATCH] back up programs after Boulder Convening
---
Index: tambo/js/soundManager.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tambo/js/soundManager.ts b/tambo/js/soundManager.ts
--- a/tambo/js/soundManager.ts	(revision ddba4c0c1b88a11599d0623a2b016925dd4c1d3e)
+++ b/tambo/js/soundManager.ts	(date 1683326031121)
@@ -244,6 +244,8 @@
         const fullyEnabled = enabled && audioEnabled && simInitComplete && simVisible && simActive && !simSettingPhetioState;
         const gain = fullyEnabled ? this._masterOutputLevel : 0;
 
+        phet.logDebugMessage( `gain = ${gain}` );
+
         // Set the gain, but somewhat gradually in order to avoid rapid transients, which can sound like clicks.
         this.masterGainNode!.gain.linearRampToValueAtTime(
           gain,
@@ -300,6 +302,7 @@
     // behavior of the audio context becomes more consistent across browsers, it may be possible to simplify this.
     //------------------------------------------------------------------------------------------------------------------
 
+    phet.logDebugMessage( `phetAudioContext.state = ${phetAudioContext.state}` );
     // function to remove the user interaction listeners, used to avoid code duplication
     const removeUserInteractionListeners = () => {
       window.removeEventListener( 'touchstart', resumeAudioContext, false );
@@ -311,19 +314,24 @@
     // listener that resumes the audio context
     const resumeAudioContext = () => {
 
+      phet.logDebugMessage( `  resumeAudioContext called, phetAudioContext.state = ${phetAudioContext.state}` );
+
       if ( phetAudioContext.state !== 'running' ) {
 
         phet.log && phet.log( `audio context not running, attempting to resume, state = ${phetAudioContext.state}` );
+        phet.logDebugMessage( `  audio context not running, attempting to resume, state = ${phetAudioContext.state}` );
 
         // tell the audio context to resume
         phetAudioContext.resume()
           .then( () => {
             phet.log && phet.log( `resume appears to have succeeded, phetAudioContext.state = ${phetAudioContext.state}` );
+            phet.logDebugMessage( `  resume appears to have succeeded, phetAudioContext.state = ${phetAudioContext.state}` );
             removeUserInteractionListeners();
           } )
           .catch( err => {
             const errorMessage = `error when trying to resume audio context, err = ${err}`;
             console.error( errorMessage );
+            phet.logDebugMessage( '  ' + errorMessage );
             assert && alert( errorMessage );
           } );
       }
Index: quadrilateral/js/quadrilateral/view/QuadrilateralScreenView.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/quadrilateral/js/quadrilateral/view/QuadrilateralScreenView.ts b/quadrilateral/js/quadrilateral/view/QuadrilateralScreenView.ts
--- a/quadrilateral/js/quadrilateral/view/QuadrilateralScreenView.ts	(revision f66075aac63e2f68aaaf163b96211037d0724e0b)
+++ b/quadrilateral/js/quadrilateral/view/QuadrilateralScreenView.ts	(date 1683325789392)
@@ -18,7 +18,7 @@
 import QuadrilateralConstants from '../../QuadrilateralConstants.js';
 import quadrilateral from '../../quadrilateral.js';
 import QuadrilateralModel from '../model/QuadrilateralModel.js';
-import { VBox } from '../../../../scenery/js/imports.js';
+import { RichText, VBox } from '../../../../scenery/js/imports.js';
 import QuadrilateralQueryParameters from '../QuadrilateralQueryParameters.js';
 import QuadrilateralNode from './QuadrilateralNode.js';
 import QuadrilateralSoundView from './sound/QuadrilateralSoundView.js';
@@ -42,6 +42,7 @@
 import QuadrilateralModelViewTransform from './QuadrilateralModelViewTransform.js';
 import QuadrilateralTangibleController from './prototype/QuadrilateralTangibleController.js';
 import { SpeakableResolvedResponse } from '../../../../utterance-queue/js/ResponsePacket.js';
+import PhetFont from '../../../../scenery-phet/js/PhetFont.js';
 
 export default class QuadrilateralScreenView extends ScreenView {
   private readonly model: QuadrilateralModel;
@@ -221,6 +222,16 @@
     //---------------------------------------------------------------------------------------------------------------
     this.pdomPlayAreaNode.pdomOrder = [ this.quadrilateralNode, shapeNameDisplay, resetShapeButton, shapeSoundsCheckbox ];
     this.pdomControlAreaNode.pdomOrder = [ visibilityControls, smallStepsLockToggleButton, resetAllButton, deviceConnectionParentNode ];
+
+
+    // TODO: Debug, don't commit.
+    const debugTextNode = new RichText( phet.debugStringProperty, {
+      x: 50,
+      y: 94,
+      font: new PhetFont( 16 ),
+      stroke: 'red'
+    } );
+    this.addChild( debugTextNode );
   }
 
   /**
Index: quadrilateral/js/quadrilateral-main.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/quadrilateral/js/quadrilateral-main.ts b/quadrilateral/js/quadrilateral-main.ts
--- a/quadrilateral/js/quadrilateral-main.ts	(revision f66075aac63e2f68aaaf163b96211037d0724e0b)
+++ b/quadrilateral/js/quadrilateral-main.ts	(date 1683325789392)
@@ -17,6 +17,7 @@
 import QuadrilateralOptionsModel from './quadrilateral/model/QuadrilateralOptionsModel.js';
 import QuadrilateralInputPreferencesNode from './quadrilateral/view/prototype/QuadrilateralInputPreferencesNode.js';
 import QuadrilateralSoundOptionsNode from './quadrilateral/view/sound/QuadrilateralSoundOptionsNode.js';
+import StringProperty from '../../axon/js/StringProperty.js';
 
 const quadrilateralTitleStringProperty = QuadrilateralStrings.quadrilateral.titleStringProperty;
 const optionsModel = new QuadrilateralOptionsModel();
@@ -60,6 +61,11 @@
   } )
 };
 
+phet.debugStringProperty = new StringProperty( '' );
+phet.logDebugMessage = ( msg: string ) => {
+  phet.debugStringProperty.value = phet.debugStringProperty.value + msg + '<br>';
+};
+
 // launch the sim - beware that scenery Image nodes created outside of simLauncher.launch() will have zero bounds
 // until the images are fully loaded, see https://github.com/phetsims/coulombs-law/issues/70
 simLauncher.launch( () => {


</details>

**Comment by jessegreenberg** [comment]
@jbphet thank you so much for looking into this and the detailed description. Ill review this with the design team to decide next steps.

I agree that escalation to a common code issue is probably best, though I might recommend it not block quadrilateral publication since it impacts several sims.

**Comment by jessegreenberg** [comment]
It is likely that this comment from @KatieWoe is related to this issue, pasting here for documentation:

```
I saw a bit of an oddity on iPad. If you start the sim but don't interact with it for a minute, when you do start to interact sound won't play if you drag something. It will start and behave normally if you tap the screen, like pushing a button. This doesn't happen if you interact right away or stop interacting for awhile after already interacting with the sim.
```

**Comment by BLFiedler** [comment]
Elevating this for PhET team priority since it is cross-sim. Does not block Quad publication.

Thanks QA! Can be ignored in next RC. Closing.
